docs/developers/best_practices.md

```diff
@@ -45,19 +45,54 @@ sample for further guidance on using dynamic shapes.
 
 ## Practices for compilation settings
 
-TODO: mention parameters to tune
-
 TODO: which compiler targets to use (try both CUDA and Vulkan?)
 
 TODO: use the most specific LLVM target triple you can?
 
-## Practices for runtime use
+### Tuning compilation heuristics
 
-### Do the minimum amount of work: cache queries and reuse buffers
+IREE runs its own suite of benchmarks continuously using the definitions at
+https://github.com/google/iree/tree/main/benchmarks. The flags set for these
+benchmarks represent the latest manually tuned values for workloads we track
+closely and referencing them may help with your own search for peak performance.
 
-Try to front-load queries, particularly queries using strings that look up into
-maps like `iree_runtime_session_call_by_name`, so that hot sections of code are
-doing the minimum amount of work: routing inputs through buffers, scheduling
-runtime calls, and routing outputs through other buffers.
+Here is a non-exhaustive list of flags which can be tuned when compiling
+through the `iree-translate` tool, while full documentation can be found in the
+project source code for each flag:
+
+* `--iree-flow-inline-constants-max-byte-length=[integer]`: Maximum byte-length of constants
+that can be inlined into dispatch regions. We find values around 16 work best
+for when running on a GPU and values around 2048 work best on CPUs.
+* `--iree-llvm-loop-unrolling=true`: This flag and other flags like it in
+[LLVMTargetOptions.cpp](https://github.com/google/iree/blob/main/iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp)
+turn on specific optimizations within LLVM when targeting CPUs.
+* `--iree-flow-dispatch-formation-enable-operand-fusion=true`: Enable fusing operand
+producers during dispatch region formation.
+* `--iree-enable-fusion-with-reduction-ops=true`: Allow fusing generic ops with
+reductions.
+
+## Practices for runtime use
 
 TODO: sample code, profile numbers
+
+### Tuning runtime settings
+
+When running on the CPU, the task system flags specified in
+[iree/task/api.c](https://github.com/google/iree/blob/main/iree/task/api.c)
+give control over how worker threads will be created. For example, the
+`--task_topology_group_count=3` flag can be set to explicitly run on three
+workers rather than rely on heuristic selection that defaults to one worker
+per detected physical core.
+
+If running on a single thread or system with no threading support, the
+`dylib-sync` HAL driver can be used instead of the more generic `dylib` HAL
+driver. The synchronous driver performs execution inline rather than through
+IREE's task scheduling system.
+
```

Comment:
It would be nice to also mention for GPU one would need to choose the matching triple to get best perf. (Can defer to a link to the main GPU page. I'll update there later.)

Reply:
There is a spot for that above, and on the website page for GPU deployment.